x-pack/test/fleet_api_integration/apis/fixtures/test_packages/non_epr_fields/1.0.0/data_stream/test_metrics/manifest.yml

```diff
@@ -13,4 +13,5 @@ streams:
       show_user: true
 
 elasticsearch:
-  source_mode: "synthetic"
\ No newline at end of file
+  # Good to see these packages for testing. Is there a flow to test upgrades?
+  source_mode: "synthetic"
```

Comment:
We have some assertions around updates in https://github.com/elastic/kibana/blob/86527753a2db307138ea69e0ba04551717a91844/x-pack/test/fleet_api_integration/apis/epm/update_assets.ts#L37, but I don't think there are specific integration tests around upgrades where `source_mode` or `index_mode` are toggled. Definitely worth adding some, especially since the package spec values will be the sole source of truth in the future and the UI toggles will be removed.